### ISSUE TITLE
CallbackMediaPlayerComponent appears to incorrectly scale video

### ISSUE THREAD

**Issue by adamish** [caprica/vlcj#1093]
I'm am playing a 1280x720 video in a 480x270 VLCJ player. This is exactly 4 times smaller in width and height. As such I would expect it to precisely fill the player.

With CallbackMediaPlayerComponent there are black bars shown, with EmbeddedMediaPlayer there are not.

![image](https://user-images.githubusercontent.com/1560724/141283351-652583c1-1074-42b1-91ce-3cadbb3dc910.png)

I've demonstrated this with the following example. The test video I used was from a stock video website https://pixabay.com/videos/download/video-83880_small.mp4?attachment

ffprobe reports this as
```
Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 1280x720, 1053 kb/s, 59.94 fps, 59.94 tbr, 60k tbn, 120k tbc (default)
```

### Example
```
import java.awt.Canvas;
import java.awt.Color;
import java.awt.Dimension;

import javax.swing.JFrame;
import javax.swing.JPanel;

import uk.co.caprica.vlcj.factory.MediaPlayerFactory;
import uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent;
import uk.co.caprica.vlcj.player.component.MediaPlayerSpecs;
import uk.co.caprica.vlcj.player.component.MediaPlayerSpecs.CallbackMediaPlayerSpec;
import uk.co.caprica.vlcj.player.embedded.EmbeddedMediaPlayer;

public class AspectRatioIssue {

    public static void main(String[] args) throws Exception {
        JFrame frame = new JFrame();
        JPanel container = new JPanel();
        frame.add(container);
        
        // callback player
        MediaPlayerFactory mediaPlayerFactory = new MediaPlayerFactory();
        CallbackMediaPlayerSpec spec = MediaPlayerSpecs.callbackMediaPlayerSpec().withFactory(mediaPlayerFactory);
        CallbackMediaPlayerComponent callbackPlayer = new CallbackMediaPlayerComponent(spec);
        callbackPlayer.setPreferredSize(new Dimension(480, 270));
        container.add(callbackPlayer);
        
        // regular player
        Canvas videoSurface = new Canvas();
        videoSurface.setBackground(Color.black);
        videoSurface.setPreferredSize(new Dimension(480, 270));
        EmbeddedMediaPlayer embeddedPlayer = mediaPlayerFactory.mediaPlayers().newEmbeddedMediaPlayer();
        embeddedPlayer.videoSurface().set(mediaPlayerFactory.videoSurfaces().newVideoSurface(videoSurface));
        container.add(videoSurface);

        frame.setVisible(true);
        frame.setSize(800, 600);
        callbackPlayer.mediaPlayer().media().play("/mnt/hgfs/Downloads/Sunrise - 83880.mp4");
        embeddedPlayer.media().play("/mnt/hgfs/Downloads/Sunrise - 83880.mp4");
    }

}
```

### Analysis

I've debugged this myself, I've found uk.co.caprica.vlcj.player.component.callback.ScaledCallbackImagePainter.paint(Graphics2D, JComponent, BufferedImage) is called with a height of 738, this then causes the width to be reduced to 468 to compensate

The 738 appears to come via uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent.newVideoBuffer(int, int) which is called initially with 720, twice, and then with 738. The call appears to come from underlying JNA so hard to debug further

I can workaround this by applying
```
.withImagePainter(new FilledCallbackImagePainter());
```

**Comment by caprica** [comment]
Interesting report, thanks.

So VLC is telling the callback media player that the video buffer has a height of 738. The callback media player allocates that buffer size (it must do so, because VLC will fill it) and renders that entire buffer every frame.

It is therefore assuming that the video buffer height matches the pixel height, which I suppose most of the time is reasonable - but not always.

While VLC is filling these extra lines, the media player should not be including them in its own video surface height, nor rendering them.

What may be possible is to work out from track information what the "real" height is that it needs to draw. Even if so, there may be a frame or two at the very start of the video where it is wrong. That is not a big deal.

It's possible something could be improved here, I'm not sure exactly what at the moment.

**Comment by caprica** [comment]
That's a really nice video by the way. Very peaceful.

**Comment by caprica** [comment]
This looks OK:
![image](https://user-images.githubusercontent.com/1385879/141342289-7fd7e818-9043-4d1e-88dc-865951456fd9.png)

The change was in ScaledCallbackImagePainter to use the actual video track height (when available) rather than the image height to calculate the scaling factor when it renders the frame.

It seems to be correct.

**Comment by caprica** [comment]
Should be resolved with 4d0769669ccf62aaaf6bf3fa0f8924e3da6d7e67.

**Comment by caprica** [comment]
Pushed a new snapshot build including this to GitHub packages and Maven Central snapshot repo.
